processor/src/decoder/mod.rs

```diff
@@ -43,7 +43,7 @@ impl Process {
         // row addr + 7.
         let child1_hash = block.first().hash().into();
         let child2_hash = block.second().hash().into();
-        let (addr, _result) = self.chiplets.merge(child1_hash, child2_hash);
+        let (addr, _result) = self.chiplets.merge_control_blocks(child1_hash, child2_hash);
```

Comment:
As mentioned in the other comment, we could probably modify the signature of `merge_control_blocks` so that it doesn't return a result here. Instead, we cold read the result in `end_join_block` and do the `debug_asserr_eq` for block hashes there.

Same could apply to the functions for other blocks below.

Reply:
Just to be clear - this is related to your comment above about removing the queued requests, right? And therefore it's a comment for the future? I agree with exploring a refactor along these lines, but I would prefer to address it in its own PR once we have baseline functionality we're confident in (and have completed tests etc).